File: functions/libsodium/variants/c/3_b64_urlsafe_char_to_byte_7.c
#include <stdint.h>

static unsigned int b64_urlsafe_char_to_byte(int c)
{
    unsigned int x = 0;
    if (c >= 'A' && c <= 'Z')
        x = c - 'A';
    else if (c >= 'a' && c <= 'z')
        x = c - 'a' + 26;
    else if (c >= '0' && c <= '9')
        x = c - '0' + 52;
    else if (c == '-')
        x = 62;
    else if (c == '_')
        x = 63;
    return x;
}


int main(){
    b64_urlsafe_char_to_byte(0);
}

File: functions/libsodium/variants/go/3_b64_urlsafe_char_to_byte_7.go
package main

func b64_urlsafe_char_to_byte(c int32) uint {
    switch true {
    case c >= 'A' && c <= 'Z':
        return uint(c - 'A')
    case c >= 'a' && c <= 'z':
        return uint(c - 'a' + 26)
    case c >= '0' && c <= '9':
        return uint(c - '0' + 52)
    case c == '-':
        return 62
    case c == '_':
        return 63
    default:
        return 0
    }
}


func main(){
    b64_urlsafe_char_to_byte(0)
}
